src/api/addresses.ts

```diff
@@ -10,64 +10,94 @@ import {
 } from '../constants';
 import { getStartPath, queue } from './utilities';
 
-export const fetchAddresses = async (
-  overrides?: GetAddressesRequestParams,
-): Promise<string[]> => {
-  return queue((client) =>
-    client
-      .getAddresses({
-        startPath: DEFAULT_ETH_DERIVATION,
-        n: MAX_ADDR,
-        ...overrides,
-      })
-      .then((addrs) => addrs.map((addr) => `${addr}`)),
-  );
+type FetchAddressesParams = {
+  n?: number;
+  startPathIndex?: number;
+};
+
+export const fetchAddresses = async (overrides?: GetAddressesRequestParams) => {
+  let allAddresses: string[] = [];
+  let totalFetched = 0;
+  const totalToFetch = overrides?.n || MAX_ADDR;
+
+  while (totalFetched < totalToFetch) {
+    const batchSize = Math.min(MAX_ADDR, totalToFetch - totalFetched);
+    const startPath = getStartPath(DEFAULT_ETH_DERIVATION, totalFetched);
+    await queue((client) =>
+      client
+        .getAddresses({
+          startPath,
+          ...overrides,
+          n: batchSize,
+        })
+        .then((addresses: string[]) => {
+          if (addresses.length > 0) {
+            allAddresses = [...allAddresses, ...addresses];
+            totalFetched += addresses.length;
```

Comment:
I think we can move the increaser to `.finally()` in case of any connection problem that could lead to an infinite loop